ansible/files/admin_api_scripts/pg_upgrade_initiate.sh

```diff
@@ -37,78 +69,143 @@ cleanup() {
         mv /var/lib/postgresql.bak /var/lib/postgresql
     fi
 
-    systemctl restart postgresql
-    sleep 10
-    systemctl restart postgresql
+    if [ -L /usr/lib/postgresql/lib/aarch64/libpq.so.5 ]; then
+        rm /usr/lib/postgresql/lib/aarch64/libpq.so.5
+    fi
 
-    for EXTENSION in "${EXTENSIONS_TO_DISABLE[@]}"; do
-        run_sql "CREATE EXTENSION IF NOT EXISTS ${EXTENSION} CASCADE;"
-    done
+    if [ "$IS_DRY_RUN" = false ]; then
+        echo "Restarting postgresql"
+        systemctl restart postgresql
+    fi
+
+    echo "Re-enabling extensions"
+    if [ -f $POST_UPGRADE_EXTENSION_SCRIPT ]; then
+        run_sql -f $POST_UPGRADE_EXTENSION_SCRIPT
```

Comment:
might be good to run these scripts through shellcheck -- it'd probably get mad about all the non-quoted variables

Reply:
Good point! Added a [shellcheck workflow](https://github.com/supabase/postgres/actions/runs/4162177961/jobs/7201028544) to help keep the admin_api_scripts folder in check